web-frame/src/app-feature/account-feature/pages/login-page/login-page.component.ts

```diff
@@ -86,11 +96,10 @@ export class LoginPageComponent implements OnInit {
 
   private onLoginError(err: unknown): void {
     this.loginForm.enable();
-    if (typeof err === 'object' && err['message']) {
-      alert('Login error!');
-      this.loginForm.setErrors({
-        submitError: err['message'],
-      });
+    if (err instanceof HttpErrorResponse) {
+      const validationErrors = getHttpResponseValidationErrors(err);
+      this.loginForm.setErrors(validationErrors);
+      this.loginForm.markAsUntouched();
```

Comment:
With these changes the forms have a very nice UX and I think that I have figured out a way to extract the feature into a directive so that it's reusable on all pages and the form container (the login page in this case) will need to implement a simple interface and the directive will be added directly to the form element.

```

export interface IAppFormContainer<S = void> {
  readonly I18N_NAMESPACE: string;

  getSubmitObservable(formGroup: FormGroup): Observable<S>;
}
```

Reply:
Created #145